Local Pods/StyledText/StyledTextTests/StyledTextBuilderTests.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  StyledTextBuilderTests.swift
+//  StyledTextTests
+//
+//  Created by Ryan Nystrom on 12/12/17.
+//  Copyright © 2017 Ryan Nystrom. All rights reserved.
+//
+
+import XCTest
+@testable import StyledText
+
+class StyledTextBuilderTests: XCTestCase {
+    
+    func test_whenOneLevel() {
+        let render = StyledTextBuilder(styledText: StyledText(text: "foo"))
+            .render(contentSizeCategory: .large)
+        XCTAssertEqual(render.string, "foo")
+    }
+
+    func test_whenAddingString() {
+        let render = StyledTextBuilder(styledText: StyledText(text: "foo"))
+            .add(text: " bar")
+            .render(contentSizeCategory: .large)
+        XCTAssertEqual(render.string, "foo bar")
+    }
+
+    func test_whenAddingAttributes() {
+        let render = StyledTextBuilder(styledText: StyledText(text: "foo", style: TextStyle(traits: .traitBold)))
+            .add(styledText: StyledText(text: " bar", style: TextStyle(traits: .traitItalic)))
+            .render(contentSizeCategory: .large)
+        XCTAssertEqual(render.string, "foo bar")
+
+        let font1 = render.attributes(at: 1, effectiveRange: nil)[.font] as! UIFont
+        XCTAssertEqual(font1.fontDescriptor.symbolicTraits, .traitBold)
+
+        let font2 = render.attributes(at: 5, effectiveRange: nil)[.font] as! UIFont
+        XCTAssertEqual(font2.fontDescriptor.symbolicTraits, .traitItalic)
+    }
+
+    func test_whenAddingAttributes_withSavingState_thenRestoring() {
+        let render = StyledTextBuilder(styledText: StyledText(text: "foo", style: TextStyle(traits: .traitBold)))
+            .save()
+            .add(styledText: StyledText(text: " bar", style: TextStyle(traits: .traitItalic)))
+            .restore()
+            .add(text: " baz")
+            .render(contentSizeCategory: .large)
+        XCTAssertEqual(render.string, "foo bar baz")
```

Comment:
@BasThomas @Sherlouk quick peek at how this API will work. I'm pretty happy w/ the direction.

- Moves the `UIApplication` API to the render function so we could call it on main, getting rid of the warning.
- Save/restore attribute state (e.g. push/pop)
- `NSAttributedString` rendered w/ passed-in content size so that the builder could be maybe once and output different sizes, like after someone changes their settings